ahoopes16/zombie-roleplaying-server
# Implement TSOA

### Summary
This PR rewrites all of the encounter routes to use [TSOA](https://tsoa-community.github.io/docs/introduction.html#goal) instead of regular Koa middleware.

TSOA provides several benefits, such as automatic generation of [OpenAPI](http://spec.openapis.org/oas/v3.0.3) specs for the API and input validation based on the types surrounding the controller functions.

I also began using [Boom](https://www.npmjs.com/package/@hapi/boom) to generate HTTP-friendly error objects.

If accepted, this PR will:

- Rewrite existing controllers to use TSOA
- Implement errors using Boom so they are HTTP friendly errors

### To Test

1. Run `yarn start`
1. Test all of the existing encounter routes
1. Check out the generated `swagger.json` file
1. You can import this directly into Postman or Insomnia to test the API while it's running
